libs/core/langchain_core/utils/function_calling.py

```diff
@@ -495,20 +495,25 @@ def convert_to_openai_tool(
 
 
 def tool_example_to_messages(
```

Comment:
This utility was added [8 months ago](https://github.com/langchain-ai/langchain/pull/18602) but is not documented. Thoughts on retroactively marking it `@beta`?

Reply:
Sure I don't mind!

hmm tool calls are also limited to BaseModel / should probably supported typed dicts

Reply:
Marked `beta`

---

Looked into supporting TypedDict but we need a way of getting the tool name from an **instance** of TypedDict

```python
from typing_extensions import TypedDict


class Apple(TypedDict):
    color: str


tool_call = Apple(color="red")
tool_call.__class__.__name__
```
```
'dict'
```